### ISSUE TITLE
Update elemental database 

### ISSUE THREAD

**Issue by Morgane21** [hyperspy/hyperspy#2723]
I would like to ask for your help concerning the EDX analysis. In the current elemental database, some M rays of Uranium are missing and do not allow me to do a correct fit and to distinguish it from Cesium, Iodine and Tellurium(which are in this energy range). Is there a solution to consider these rays in order to have a better identification of the elements? 
Thank you in adavance

![image](https://user-images.githubusercontent.com/83064111/116217845-427f7580-a74a-11eb-87b3-574c1dbd7035.png)

**Comment by ericpre** [comment]
Thanks for opening the issue. You could already add manually these components and use the `twin_function_expr` functionality to fix the ratio between lines but this is a bit tedious - this is what the EDS model is setting up the components.

Maybe, an alternative would be to be able to pass your own database as a parameter when creating the EDS model?

Something along the line of:
```python
import hyperspy.api as hs
from hyperspy.misc.elements import elements as elements_db

my_custom_lines = {'Mz2': {'energy (keV)': 2.455, 'weight':0.017}}
elements_db['U']['Atomic_properties']['Xray_lines'].update(my_custom_lines)

s = hs.datasets.example_signals.EDS_TEM_Spectrum()
m = s.create_model(element_db=elements_db)
```

@sem-geologist, @TomSlater, @k8macarthur, in case you would like to comment on this.

**Comment by Morgane21** [comment]
I tried but I get an error message when I run: m = s.create_model(element_db=elements_db). I don't know where the problem comes from  
![image](https://user-images.githubusercontent.com/83064111/116458027-6c2cbf80-a864-11eb-8410-7df993333918.png)

**Comment by thomasaarholt** [comment]
@Morgane21 eric's suggestion is to introduce new functionality to support that syntax. That would require someone to code a way to do that in a new Pull Request here on github :)

**Comment by francisco-dlp** [comment]
I wonder if such a new functionality may be counterproductive since it will encourage people to workaround the issue rather than opening an issue like @Morgane21 has done.

Why not simply updating the database with the values in the article?

**Comment by thomasaarholt** [comment]
I agree with @francisco-dlp - though it would be important to make sure that the existing weights match the new ones.

**Comment by ericpre** [comment]
See @sem-geologist comment on gitter https://gitter.im/hyperspy/hyperspy?at=6085f0fca2ac0d38e7dacc46 for more context.

The database most likely works fine for many many application and very few users will need to change the default values. @Morgane21 may be providing some weights working fine for themselves but it doesn't mean that other users looking at this level of accuracy will be happy with these values, or any other of these defaults values! They will most likely have the need to fiddle/calibrate these values themselves and therefore we should provide a more generic and flexible solution.

In any case, it would be good to add these values to the database.

**Comment by francisco-dlp** [comment]
It seems that there are 2 problems here:

i) the database is incomplete
ii) there is no consensus on the position and relative intensity of the peaks at the accuracy required for quantification


So, I agree with @ericpre, it would be good to fix i by adding the missing values to the database.

Regarding ii, @sem-geologist, how can we improve the situation?

**Comment by sem-geologist** [comment]
I see two planes of this to discuss, software implementation and physical (xray-line-weight).

What is this line weight? What does it represent in the Hyperspy and is it used in anything apart of showing line markers on the chart. If it is used for modeling, deconvolution, and processing of spectra, then wrong weight value can introduce significant error.

Looking through different kind of software I see that there is two kind of approaches: most common is like in the hyperspy - the fixed value approach, weights of lines are hard coded. Then there are two subways how to handle differences between the line families (K,L,M): 1) normalizing intensities to highest intensity in the line as 1 (or 100, if dealing in percents as in  the screenshot above); 2) preserve measured weights/intensities in between line families (example 'xraydb'). But what is the line weight? it is probability of emission at given energy, which depends from available electron configuration in given atom, and applied energy to excite the atom. All to me known packages treats probabilities as fixed values, and ignoring the meaning of that and how does line weight is used (and overused) in given software, works reasonable well while working with K and L lines....
But M lines are something specific, especially for elements with Z < 70 (Ytterbium). The absorption edge energies is converging with M alpha and beta emission energies going toward lower atom number.  The effect is so severe, that if to take high resolution observed spectra literally, there is many lines observed instead of Ma and Mb. It was long ago demonstrated that those observed lines are residual from self-absorption following exactly absorption distribution (see Fischer and Baun, 1967). So many software packages for LREE places some other MxNy lines while giving near 0 weight for Ma lines. I.e. Bruker gives ~0 weight for La Ma; DTSA-II shows no La Ma marker at all, Ce starts from Ma2 (and Cs and Ba are stripped of Ma too); NIST's NeXL goes even further - it checks with theoretical possible electron configurations and forbid transition lines if there is no theoretical electron configuration for that - which results in absurd of complete Ma (or N5-M7,6 transitions in UIPAC notation) lines missing for any element lighter than Gd.

Where everyone takes line weights from? 'pyxray' (python library for xray transitions written in java'ish style) i.e. documented that it took data from DTSA-II - it is, however, clearly modified (which is not mentioned) as it includes weights 1.0 for Ma lines starting with Lanthanium. As author of 'pyxray' works with electron microprobes - the modification is clear, the observed real existence of these lines can't be taken away by some theoretical blabery. But unfortunately, 'pyxray' dataset does not include Ma lines for Cs, Ba and lower atom numbers. When DTSA-II uses its epq library, and Hyperspy states that uses line-weights from 'epq'. But Hyperspy clearly has some modifications, (as La Ma line with weight of 1), while not including lines with very small weight.
And then there is 'xraydb', which recognize existence of  most of lines, however all energy positions have weird offsets (imprecision probably does not matter for EDX, as library comes from XRF realm, but is eye-burning for WDS spectral precision), and there is plenty of wrong line weights with historical errors introduced (i.e. U Mb being higher than U Ma, the error caused by unaccounted absorption of Ma by Argon (gas used in gas proportional flow counter) during some weight measurement attempts).

To summarize, The existing/available commercial and OSS xray line weight databases are a huge mess, there is not a single database which would be satisfactory for all possible cases.

-------
Now technical side.
I like 'xraydb'`s approach with SQLite database. The hardcoded dictionary in Hyperspy looks reasonable OK, and is good for performance of hyperspy, however adding data or refining values is not so user friendly. Another point, I think most often this is a waste of resources, as at start of Hyperspy all database needs to be loaded, while in most cases only set of elements is going to be used. I would suggest to consolidate data into SQLite database, and load and build the dictionary lazily for requested elements (the dictionary would be like caching here). That would increase a bit (talking about few ms) of overhead at first element request, but would be same as now fast for later usage. When requesting lines from database, we could define default weight filter, so that it would not clutter plotting, and would behave same as now. However, when someone would need all lines, it would be enough to lower weight filter, either globally or per element basis.

**Comment by sem-geologist** [comment]
I just want to mention the DTSA-II (and NeXL) approach which works (and does not work). To overcome line weight and shape dilema, DTSA-II requires reference peaks - the spectra with unobstructed/ not overlapped by other element lines. It works well unless DTSA-II negates very existence of some lines - it then does not recognize particular lines in reference spectra, and thus fits nonsense for that region of spectra.

**Comment by sem-geologist** [comment]
I see `xraydb` had fixed some of my mentioned issues above. I am going to investigate how precise those lines are against my WDS sets of ~70 elements (Z from 9 to 92, except inert gases and (Po, At, Fr, Ra, Ac, Pa)). I will write update here. It is too early to get excited.
